src/shared/dissect-image.c

```diff
@@ -150,6 +150,43 @@ static void check_partition_flags(
 }
 #endif
 
+#if HAVE_BLKID
+static int dissected_image_new(const char *path, DissectedImage **ret) {
+        _cleanup_(dissected_image_unrefp) DissectedImage *m = NULL;
+        _cleanup_free_ char *name = NULL;
+        int r;
+
+        assert(ret);
+
+        if (path) {
+                _cleanup_free_ char *filename = NULL;
+
+                r = path_extract_filename(path, &filename);
+                if (r < 0)
+                        return r;
+
+                r = raw_strip_suffixes(filename, &name);
+                if (r < 0)
+                        return r;
+
+                if (!image_name_is_valid(name))
+                        log_debug("Image name %s is not valid, ignoring.", strna(name));
```

Comment:
We say we ignore the filename, but we just use it as if we'd gotten a valid filename, that doesn't seem entirely right, shouldn't we drop `name` if this happens

Reply:
Ouch, indeed. Thank you.